### ISSUE TITLE
Apps crashed when I click quick on any of resources bar. 

### ISSUE THREAD

**Issue by kiltum** [kontena/lens#3972]
Update: it happens every time when system busy and cannot process my request fast enough 

Simple one node local k8s (created from docker desktop). Nothing special at all
```
Lens: 5.2.5-latest.20211001.2
Electron: 12.2.1
Chrome: 89.0.4389.128
Node: 14.16.0
```
kubectl get cronjobs
```
NAME        SCHEDULE    SUSPEND   ACTIVE   LAST SCHEDULE   AGE
retention   0 0 * * *   False     0        <none>          6m5s
```
describe
```
Name:                          retention
Namespace:                     luntry
Labels:                        <none>
Annotations:                   <none>
Schedule:                      0 0 * * *
Concurrency Policy:            Allow
Suspend:                       False
Successful Job History Limit:  7
Failed Job History Limit:      7
Starting Deadline Seconds:     <unset>
Selector:                      <unset>
Parallelism:                   <unset>
Completions:                   <unset>
Pod Template:
  Labels:  <none>
  Containers:
   retention:
    Image:      registry.luntry.com/luntry/retention:develop
    Port:       <none>
    Host Port:  <none>
    Environment:
      POSTGRES_HOST:                  <set to the key 'POSTGRES_HOST' in secret 'luntry-config'>                  Optional: false
      POSTGRES_PORT:                  <set to the key 'POSTGRES_PORT' in secret 'luntry-config'>                  Optional: false
      POSTGRES_DB:                    <set to the key 'COLLECTOR_DATABASE' in secret 'luntry-config'>             Optional: false
      POSTGRES_USER:                  <set to the key 'RETENTION_USER' in secret 'luntry-config'>                 Optional: false
      POSTGRES_PASSWORD:              <set to the key 'RETENTION_PASSWORD' in secret 'luntry-config'>             Optional: false
      ANOMALY_RETENTION:              <set to the key 'ANOMALY_RETENTION' in secret 'luntry-config'>              Optional: false
      ANOMALY_OBSERVATION_RETENTION:  <set to the key 'ANOMALY_OBSERVATION_RETENTION' in secret 'luntry-config'>  Optional: false
      KUBERNETES_HISTORY_RETENTION:   <set to the key 'KUBERNETES_HISTORY_RETENTION' in secret 'luntry-config'>   Optional: false
      POLICY_RETENTION:               <set to the key 'POLICY_RETENTION' in secret 'luntry-config'>               Optional: false
      EVENTS_RETENTION:               <set to the key 'EVENTS_RETENTION' in secret 'luntry-config'>               Optional: false
    Mounts:                           <none>
  Volumes:                            <none>
Last Schedule Time:                   <unset>
Active Jobs:                          <none>
Events:                               <none>
```

Component stack:

```
    at InstallChart (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:663538:9)
    at div
    at div
    at Dock (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:665167:9)
    at div
    at div
    at MainLayout (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:572652:9)
    at ErrorBoundary (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:572113:47)
    at ErrorBoundary (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:572222:36)
    at Router (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:32134:30)
    at App (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:666393:9)
    at ThemeProvider (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:700069:24)
```

Error stack:

```
TypeError: Cannot read property 'values' of undefined
    at InstallChart.get values [as values] (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:663572:31)
    at InstallChart.render (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:663616:35)
    at allowStateChanges (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:1472:12)
    at http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:10151:59
    at trackDerivedFunction (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:2024:18)
    at Reaction.track (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:2552:18)
    at InstallChart.reactiveRender [as render] (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:10149:14)
    at InstallChart.makeComponentReactive (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:10164:25)
    at InstallChart.target.render (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:10077:34)
    at qi (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535680:188)
    at pi (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535679:173)
    at ck (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535762:427)
    at bk (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535743:347)
    at ak (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535743:278)
    at Tj (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535743:138)
    at Lj (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535736:163)
    at http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535616:115
    at exports.unstable_runWithPriority (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535827:343)
    at gg (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535615:325)
    at jg (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535616:61)
    at ig (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535615:428)
    at Wj (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:535737:190)
    at reactionScheduler (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:2679:12)
    at http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:2680:14
    at Gh (C:\Users\multi\AppData\Local\Programs\Lens\resources\extensions\lenscloud-lens-extension\dist\renderer.js:347:141185)
    at reactionScheduler (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:2679:12)
    at runReactions (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:2648:3)
    at endBatch (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:2315:5)
    at _endAction (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:1461:3)
    at executeAction (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:1414:5)
    at storageHelper_StorageHelper.merge (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:1393:12)
    at value (http://d5e549a6523966e8f76c6f29c53a4d3d.localhost:55574/build/Lens.js:7191:26)
```

**Comment by qayshp** [comment]
I just had a crash with very similar (edit: similar to my eyes, perhaps not?) stacks for both component and error, but haven't been able to repro it, and I'm not sure if I should open a new issue or just add here.

I had just clicked on a hotbar icon for a disconnected AWS EKS cluster (B) and the crash happened shortly after the _authentication proxy started_ notification. Before clicking on that hotbar icon, I was viewing a different EKS cluster (A) with the Resource Map extension (installed and) actively selected. The map for cluster A had not yet rendered as of then.

```
Lens: 5.2.5-latest.20211001.2
Electron: 12.2.1
Chrome: 89.0.4389.128
Node: 14.16.0
```

> App crash at /cluster
> To help us improve the product please report bugs to Slack community or GitHub issues tracker.

Component stack:
```
    at InstallChart (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:663640:9)
    at div
    at div
    at Dock (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:665269:9)
    at div
    at div
    at MainLayout (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:572754:9)
    at ErrorBoundary (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:572215:47)
    at ErrorBoundary (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:572324:36)
    at Router (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:32134:30)
    at App (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:666495:9)
    at ThemeProvider (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:700171:24)
```
Error stack:
```
TypeError: Cannot read property 'values' of undefined
    at InstallChart.get values [as values] (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:663674:31)
    at InstallChart.render (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:663718:35)
    at allowStateChanges (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:1472:12)
    at http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:10151:59
    at trackDerivedFunction (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:2024:18)
    at Reaction.track (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:2552:18)
    at InstallChart.reactiveRender [as render] (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:10149:14)
    at InstallChart.makeComponentReactive (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:10164:25)
    at InstallChart.target.render (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:10077:34)
    at qi (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535680:188)
    at pi (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535679:173)
    at ck (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535762:427)
    at bk (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535743:347)
    at ak (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535743:278)
    at Tj (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535743:138)
    at Lj (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535736:163)
    at http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535616:115
    at exports.unstable_runWithPriority (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535827:343)
    at gg (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535615:325)
    at jg (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535616:61)
    at ig (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535615:428)
    at Wj (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:535737:190)
    at reactionScheduler (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:2679:12)
    at http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:2680:14
    at Gh (/Applications/Lens.app/Contents/Resources/extensions/lenscloud-lens-extension/dist/renderer.js:347:141185)
    at reactionScheduler (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:2679:12)
    at http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:2680:14
    at eu (/Users/qpoonawala/.k8slens/extensions/lens-debug-tools/dist/renderer.js:9:93936)
    at reactionScheduler (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:2679:12)
    at runReactions (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:2648:3)
    at endBatch (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:2315:5)
    at _endAction (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:1461:3)
    at executeAction (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:1414:5)
    at storageHelper_StorageHelper.merge (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:1393:12)
    at value (http://695a422aee13a9d69c3aceed48628984.localhost:50993/build/Lens.js:7191:26)
```

**Comment by Nokel81** [comment]
Thanks for this report, this seems to be a race condition.

**Comment by kiltum** [comment]
I just reproduce this behavior. Just press quickly some times on any bar, when system is under heavy load and this happens. Close program, reopen, click quickly - again.

**Comment by Nokel81** [comment]
PR is up with fix

**Comment by ixrock** [comment]
@Nokel81 why did you closed this issue? #4225 did the job?  Did you checked?
I think that part about `install-chart` still left in original PR (#3910)

**Comment by Nokel81** [comment]
@ixrock Sorry my bad I though this was a duplicate of the other issue. I will reopen and remove the link

**Comment by Nokel81** [comment]
Fixed by #5841
